### DISCUSSION TITLE
Trouble autograding submissions

### DISCUSSION THREAD

**Discussion by nihaar-c** [ucbds-infra/otter-grader#858] [Q&A] [answered]
Hi, I'm having issues with autograding student notebooks. Here's what my resulting csv looks like after calling otter grade.
![image](https://github.com/user-attachments/assets/40e6c9a2-a115-442a-8880-b401660397fa)

I tried the same process on the demo assignment in the docs, and that seems to work fine. Is there anything that I should look for that might be causing this issue? I've answered every question in the student notebooks and ran every cell -- I've shared the solutions notebook if anyone could help.
[test_assignment-sol.pdf](https://github.com/user-attachments/files/17296371/test_assignment-sol.pdf)

Thanks

**Comment by nihaar-c** [comment]
Did some debugging and found that the root cause is `!git clone https://github.com/zpqrtbnk/test-repo.git`. Is Otter not compatible with cmd line commands? Would that be something that messes up the grading workflow?

**Reply by chrispyles** [reply, marked as answer]
Otter should work fine with shell commands (assuming you're using a recent version) so I'm not sure what could be causing this. You can see what error is occurring in the notebook by following [these debugging instructions](https://otter-grader.readthedocs.io/en/latest/debugging.html). Then, after you run `./run_autograder` in the container, from outside the container run `docker cp <container ID>:/autograder/results/results.pkl ./results.pkl` then run the following python script to extract the notebook:

```python
import nbformat
import dill

with open("results.pkl", "rb") as f:
    res = dill.load(f)

nbformat.write(res.notebook, "executed.ipynb")
```

Open the notebook and find the cell with the shell command, and the error should be in its output.
